pkg/indicator/alma.go

```diff
@@ -0,0 +1,95 @@
+package indicator
+
+import (
+	"math"
+
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+// Refer: Arnaud Legoux Moving Average
+// Refer: https://capital.com/arnaud-legoux-moving-average
+// Also check https://github.com/DaveSkender/Stock.Indicators/blob/main/src/a-d/Alma/Alma.cs
+// @param offset: Gaussian applied to the combo line. 1->ema, 0->sma
+// @param sigma: the standard deviation applied to the combo line. This makes the combo line sharper
+//go:generate callbackgen -type ALMA
+type ALMA struct {
+	types.IntervalWindow         // required
+	Offset               float64 // required: recommend to be 5
+	Sigma                int     // required: recommend to be 0.5
+	Weight               []float64
+	Sum                  float64
+	input                []float64
+	Values               types.Float64Slice
+	UpdateCallbacks      []func(value float64)
+}
+
+const MaxNumOfALMA = 5_000
+const MaxNumOfALMATruncateSize = 100
+
+func (inc *ALMA) Update(value float64) {
```

Comment:
call this `update` and rename `calculateAndUpdate` to `Update` as an exported method?